source/docs/Sensors/limit-switches.rst

```diff
@@ -36,3 +36,40 @@ Reading a Digital Input
 
             //Use these to access data
             input.Get(); //Will return true for a high signal and false for a low signal
+            
+    .. tab:: Roscpp
+    
+         .. code-block:: c++
+            :linenos:
+            
+            //Include the DigitalInput and the DigitalOuptut Library
+            #include "DI_ros.h"
+            #include "DI_ros.h"
+            
+            /**
+             * Constructors
+             * Create the DigitalOuptut and DigitalInput objects
+             * DIO ros threads (publishers and services) will run asynchronously in the background
+             */
+            
+            ros::NodeHandle nh; //internal reference to the ROS node that tourm program will use to interacts with the ROS system
+            VMXPi vmx(true, (uint8_t)50); //realtime bool and the update rate to use for the VMXPi AHRS/IMU interface, defalut is 50hz within a valid range of 4-200Hz
+            
+            DigitalOuptutROS(&nh, &vmx, channel);
+            DigitalInputROS(&nh, &vmx, channel);
+            
+            ros::ServiceClient sendPing, setDuration;
+            
+            // Declaring message types
+            std_srvs::Empty msg1; //ROS service type for sending a signal to a ROS node
+            vmxpi_ros::Int msg2; //VMX-pi service type
+            
+            //Use these to send a ping signal and set duration
```

Comment:
commenting here and below can be improved as it states the same thing over and over